src/main/java/ether/TransactionsManager.java

```diff
@@ -0,0 +1,663 @@
+package ether;
+import io.left.rightmesh.id.MeshID;
+import io.left.rightmesh.mesh.MeshManager;
+import io.left.rightmesh.proto.MeshTransaction;
+import io.left.rightmesh.util.EtherUtility;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.concurrent.*;
+import static io.left.rightmesh.mesh.MeshManager.TRANSACTION_RECEIVED;
+
+
+/**
+ * The Transactions manager, responsible to process incoming Ether Transactions from Clients in the Mesh network.
+ */
+public final class TransactionsManager {
+
+    private static final int NUM_OF_THREADS = 100;
+
+    private MeshManager meshManager;
+    private LinkedBlockingQueue<MeshManager.MeshTransactionEvent> transactionsQueue = new LinkedBlockingQueue<>();
+    private ConcurrentHashMap<String, EtherUtility.PaymentChannel> channelsMap = new ConcurrentHashMap<>();
+    private ExecutorService executor = Executors.newFixedThreadPool(NUM_OF_THREADS);
+    private Thread queueThread = null;
+    private volatile boolean isRunning = false;
+    private Http httpAgent;
+    private MeshID ownMeshId;
+
+
+    private static volatile TransactionsManager instance = null;
+
+    public static TransactionsManager getInstance(MeshManager mm) {
```

Comment:
Why is this a singleton? It appears to be only created in the one place in the codebase.

Singletons are tricky to unit test and we're definitely going to want them for this codebase  in the future.

Reply:
It's a good question. I just wanted to ensure that there will be only one instance of this Object, no matter how and where we are going to use it. As, this Object manages the Transactions queue and it's thread, and register itself as a Transaction handler to the MeshManager. Therefore, if somehow multiple instances will be created it will be a mess. If we can ensure that only one instance will be created during it's use, than it's OK to change it to be not a Singleton.

Reply:
Makes sense. I suppose we can leave it for now and change it if/when we right tests for this codebase.

Are there other places in the code where you can envision us needing another instance of this class?

Reply:
Not right now, lets reconsider how we would like to use it later. We could switch easily to the none singleton version, if we would like to.